app.py

```diff
@@ -1,76 +1,94 @@
 from flask import Flask, render_template, request
 from flask_socketio import SocketIO
-#from camera import Camera
+import time
+from bot import Bot
 
-import os
 
-app = Flask(__name__)
-app.config['SECRET_KEY'] = 'sennagang_pkpap'
-socketio = SocketIO(app)
-
-# intialising client ip
+# initialising client ip
 CLIENT_IP = "0.0.0.0"
 
 # list of valid user commands
 VALID_USER_COMMANDS = ['left', 'right', 'up', 'down']
 
 
 def handle_user_command(cmd):
-    ''' Function to handle valid user chat commands
-        TBD: Based on user commands, drive motor (i.e change Pi GPIO outputs)
-    '''
-    print "Command: {}".format(cmd)
+    """Function to handle valid user chat commands. Based on user commands, drive bot in that direction"""
+
+    print "User command is : {}".format(cmd)
+    cmd = str(cmd).lower()
+
+    if cmd not in VALID_USER_COMMANDS:
+        print "Invalid command : {}".format(cmd)
+        return -1
+
+    if cmd == "up":
+        bot.move_forward()
+    elif cmd == "down":
+        bot.move_backward()
+    elif cmd == "left":
+        bot.turn_left()
+    elif cmd == "right":
+        bot.turn_right()
+    else:
```

Comment:
Is this else case reachable in any scenario ? (I ask because of the check made above which returns -1 if not in the VALID command list)
If it is reachable, then shouldn't it be returning -1, so it can be handled accordingly by the user of this function ? 
@scamper07

Reply:
Yes. You are right. This else case will never be reachable. Added out of convention for if elif else. I should have either this else or the above check. Which one would be better?@Vatican-Cameos

Reply:
I would prefer to have the check initially, because, if the command is invalid it won't proceed further to check for all the ifs which is going to be redundant. 
You can keep the else case for convention, if required. If you decide to keep it, make sure you return -1 again just out of convention.
We can also return an integer, for example : 0, in case if it was a success. 
This allows the user of the class to take necessary actions based on this information.

Reply:
Addressed comment.
Kept both the changes and added return -1 in else